src/bin/pg_dump/pg_backup_archiver.c

```diff
@@ -3147,7 +3152,7 @@ _doSetFixedOutputState(ArchiveHandle *AH)
 
 	/* Select the role to be used during restore */
 	if (ropt && ropt->use_role)
-		ahprintf(AH, "SET ROLE %s;\n", fmtId(ropt->use_role));
+		// ahprintf(AH, "SET ROLE %s;\n", fmtId(ropt->use_role));
```

Comment:
So this was the line in question. Somehow `ropt->use_role` has an invalid value, which leads to segfault, and somehow this is the effect of my patch 🤔

Reply:
Test still fail BTW 😢

Reply:
Fixed